### PR TITLE
Resuable AddressForm

### PR DESCRIPTION
**Pull request by akosipc**
## PR Summary
- Adds `AddressForm` React component with Storybook and Test
- Integrates `AddressForm` in `BuyerForm` and integrates it to the `BuyerForm`'s FormSchema

## References
https://www.notion.so/ab967bca42c04316ba5777883cf4d615?v=181e1dff6685483ab0b9a7576fa5c6b3&p=b307360d761f4fa69b86712eecabf4a1&pm=s

## Discoveries (Optional)
In the future, this approach would be beneficial to for separating and reusing Form compoentns

## Further Checklist
- [x] I have performed a self-review on my own code.
- [x] I have included tests in my change.
- [x] I have made corresponding changes to the documentation if any apply.
- [x] I have posted this change request in our Slack #development channel

### CODE REVIEW

File: src/components/AddressForm/AddressForm.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    city: String,
+    line1: String,
+    line2?: String,
+    region: String,
+    zipCode: Number
+}
+
+export type AddressFormProps = {
+    register: UseFormRegister<Buyer>
+}
```
Comment (gdperalta) [inline src/components/AddressForm/AddressForm.tsx:28]: For reusability of this component for other users, I'm assuming we just need to create a union type here? (e.g. User, Supplier, etc)

Comment (akosipc) [reply src/components/AddressForm/AddressForm.tsx:28]: @gdperalta Yes, when we need to add those types.

**Review by gdperalta — APPROVED** [review]
reviewed and approved

**Comment by akosipc** [comment]
🤳

**Pull request merged**